cmd/commands/server/factory.go

```diff
@@ -79,16 +79,16 @@ func NewCommandWith(
 			)
 		},
 
-		sessionManagerFactory: func(vpnServerIP string) session.Manager {
-			clientConfigGenerator := openvpn.NewClientConfigGenerator(options.DirectoryRuntime, vpnServerIP, options.Protocol)
+		sessionManagerFactory: func(primitives *tls.Primitives, vpnServerIP string) session.Manager {
+			clientConfigGenerator := openvpn.NewClientConfigGenerator(primitives, vpnServerIP, options.OpenvpnPort, options.Protocol)
 
 			return openvpn_session.NewManager(
-				clientConfigGenerator,
+				openvpn_session.ServiceConfigProvider(clientConfigGenerator),
 				&session.UUIDGenerator{},
 			)
 		},
-		vpnServerFactory: func(manager session.Manager, serviceLocation dto.Location, providerID identity.Identity, callback state.Callback) *openvpn.Server {
-			serverConfigGenerator := openvpn.NewServerConfigGenerator(options.DirectoryRuntime, serviceLocation, providerID, options.Protocol)
+		vpnServerFactory: func(manager session.Manager, primitives *tls.Primitives, callback state.Callback) *openvpn.Server {
```

Comment:
Too long lines :/

Reply:
Reformated :) however a notice added to TODO box - cleanup server factory